docarray/array/storage/elastic/backend.py

```diff
@@ -0,0 +1,154 @@
+from elasticsearch.helpers import bulk
+
+if TYPE_CHECKING:
+    from ....types import (
+        DocumentArraySourceType,
+    )
+
+
+def _sanitize_table_name(table_name: str) -> str:
+    ret = ''.join(c for c in table_name if c.isalnum() or c == '_')
+    if ret != table_name:
+        warnings.warn(f'The table name is changed to {ret} due to illegal characters')
+    return ret
+
+
+@dataclass
+class ElasticConfig:
+    n_dim: int  # dims  in elastic
+    basic_auth: Optional[Tuple[str, str]] = None
+    ca_certs: Optional[str] = None
+    distance: str = 'cosine'  # similarity in elastic
+    host: Optional[str] = field(default='http://localhost')
+    port: Optional[int] = field(default=9200)
+    index_name: Optional[str] = field(default='index_name')
+    serialize_config: Dict = field(default_factory=dict)
+
+
+class BackendMixin(BaseBackendMixin):
+    """Provide necessary functions to enable this storage backend."""
+
+    def _init_storage(
+        self,
+        _docs: Optional['DocumentArraySourceType'] = None,
+        config: Optional[Union[ElasticConfig, Dict]] = None,
+        **kwargs,
+    ):
+
+        if not config:
+            raise ValueError('Empty config is not allowed for Elastic storage')
+        elif isinstance(config, dict):
+            config = dataclass_from_dict(ElasticConfig, config)
+
+        self._index_name_offset2id = 'index_offset2id'
+        self._config = config
+
+        self._client = self._build_client(self._config)
+        self._build_offset2id_index(self._index_name_offset2id)
+
+        super()._init_storage()  # CALLS _load_offset2ids > _get_offset2ids_meta
+
+    def _build_offset2id_index(self, index_name):
+        self._client.indices.delete(index=index_name, ignore=[404])
+        self._client.indices.create(index=index_name)
+
+    def _build_hosts(self, elastic_config):
+        return elastic_config.host + ':' + str(elastic_config.port)
+
+    def _build_schema_from_elastic_config(self, elastic_config):
+        da_schema = {
+            "mappings": {
+                "dynamic": "true",
+                "_source": {"enabled": "true"},
+                "properties": {
+                    "embedding": {
+                        "type": "dense_vector",
+                        "dims": elastic_config.n_dim,
+                        "index": "true",
+                        "similarity": elastic_config.distance,
+                    },
+                },
+            }
+        }
+        return da_schema
+
+    def _build_client(self, elastic_config):
+
+        client = Elasticsearch(
+            hosts=self._build_hosts(elastic_config),
+            ca_certs=elastic_config.ca_certs,
+            basic_auth=elastic_config.basic_auth,
+        )
+
+        schema = self._build_schema_from_elastic_config(elastic_config)
+
+        if client.indices.exists(index=elastic_config.index_name):
+            client.indices.delete(index=elastic_config.index_name)
```

Comment:
let's align this behaviour with other backends ?
if elastic_config.index_name is not specified (None), generate a random name and create index, otherwise, if it's specified, just load it, no need to alter the data (otherwise, there will be no persistence at all right ?)